content/blog/is-dns-migration-a-tricky-affair.md

```diff
@@ -0,0 +1,106 @@
+# Need for DNS migration
+One of our clients wanted us to move their workload from docker swarm to Amazon ECS. Earlier, they were using HaProxy to load-balance the traffic. This setup also included generation and installation of SSL certificates using in-house scripts, which the client needed to maintain. So, we decided to go with a managed solution to reduce the management overhead and chose AWS ALB and ACM combination. In order for ACM to auto-renew expiring SSL certificates, we decided to use Route53 as the DNS service, ditching GoDaddy's DNS service. BTW, if you are curious, the domain was registered with GoDaddy and hence their DNS service.
+
+These decisions landed us onto the supposedly **_EASY_** task of DNS migration. This was my first DNS migration task, and I was equally afraid as I was excited. Just then, I overheard someone (friend/colleague) saying with a very convincing tone that DNS migration is an easy task. Well, I was somewhat convinced. 😅
+
+Still, as a good engineer 🤭, I went ahead and did my research regarding DNS migration and carved out a migration plan.  Looking at the plan for migration, even my instincts were convinced. 
+
+
+![Instinct calling](/images/blog/is-dns-migration-a-ticky-affair/do_it.gif)
+# Migration Plan
+Basically I divided the migration plan in three phases along with a rollback strategy:
+## Pre-execution phase
+In this phase we exported the [zone file](https://en.wikipedia.org/wiki/Zone_file) from GoDaddy DNS service which we will be referring to for the list of domain records to migrate.
+
+Another thing we did was to reduce the [TTL](https://www.cloudflare.com/learning/cdn/glossary/time-to-live-ttl/) of all the records in GoDaddy DNS to the least possible value. These tasks we performed a day before the execution phase. For us, the maximum value of TTL was one day, so 24 hours was enough for us to reflect new TTL values.
+
+## Execution Phase
+This phase included changes at Route53's end. This included:
+1. Addition of hosted zone in Route53 for the [apex domain](https://help.easyredir.com/en/articles/453072-what-is-a-domain-apex)
+2. Addition of all the records we have in the zone file to Route53 hosted zone we just created.
+3. Update the nameserver records provided by Route53 in the domain registration configuration. This will switch the DNS service to Route53 once the [TTL](https://www.cloudflare.com/learning/cdn/glossary/time-to-live-ttl/) is expired.
+We actually used Terraform to add the hosted zone as well as other records, as this would ease up the maintenance of the records.
+
+## Verification Phase
+This phase included verification of the application and mail server. One way to verify this is by simply heading to the application domain and sending a mail to the colleague. This can be considered as a correct way to verify since we have taken care of the TTL, although, to make sure the DNS query result is being served by Route53 nameservers we have to make use of a linux utility called - 
```

Comment:
See if this reword makes sense?

s/This phase included verification of the application and mail server/This phase included verification of DNS records of our application and mail servers/g

Reply:
Break before *although*? Can start another sentence.

Reply:
Fixed in 15acecd